web/src/components/RemoteContent.tsx

```diff
@@ -129,20 +202,69 @@ const RemoteContent = ({
     [onInternalLinkClick],
   )
 
+  const onUpdateCookie = useCallback(
+    (source: string): void => {
+      externalSourcePermissions[source] = true
+      updateCookie(JSON.stringify(externalSourcePermissions), '/', window.location.hostname)
+    },
+    [externalSourcePermissions, updateCookie],
+  )
+
   useEffect(() => {
     if (!sandBoxRef.current) {
       return
     }
-    const collection = sandBoxRef.current.getElementsByTagName('a')
+    const currentSandBoxRef = sandBoxRef.current
+    const collection = currentSandBoxRef.getElementsByTagName('a')
     Array.from(collection).forEach(node => {
       if (HIJACK.test(node.href)) {
         node.addEventListener('click', handleClick)
       }
     })
-  }, [html, handleClick, sandBoxRef])
+
+    const iframes = currentSandBoxRef.getElementsByTagName('iframe')
+    Array.from(iframes).forEach((iframe: HTMLIFrameElement, index: number) => {
+      if (iframe.src !== IFRAME_BLANK_SOURCE) {
+        preserveIFrameSourcesFromContent(index, iframe.src, setContentIframeSources, contentIframeSources)
```

Comment:
Why do we need to save the sources separately except from the cookie? This looks like duplicated state. Or is there any reason why we need to map sources to the iframe index?